libraries/Microsoft.Bot.Builder.LanguageGeneration/Evaluator.cs

```diff
@@ -251,10 +320,43 @@ private string EvalMultiLineText(string exp)
         {
             // remove ``` ```
             exp = exp.Substring(3, exp.Length - 6);
-            var reg = @"@\{[^{}]+\}";
-            var evalutor = new MatchEvaluator(m => EvalExpression(m.Value));
+            return EvalTextContainsExpression(exp);
+        }
+
+        private string EvalTextContainsExpression(string exp)
+        {
+            var evalutor = new MatchEvaluator(m => EvalExpression(m.Value).ToString());
+            return Regex.Replace(exp, ExpressionRecognizePattern, evalutor);
+        }
+
+        private JToken EvalText(string exp)
+        {
+            if (string.IsNullOrEmpty(exp))
+            {
+                return exp;
+            }
+
+            if (IsPureExpression(exp))
+            {
+                // @{} or {} text, get object result
+                return JToken.FromObject(EvalExpression(exp));
+            }
+            else
+            {
+                return Regex.Unescape(EvalTextContainsExpression(exp));
+            }
+        }
+
+        private bool IsPureExpression(string exp)
+        {
+            if (string.IsNullOrWhiteSpace(exp))
+            {
+                return false;
+            }
 
-            return Regex.Replace(exp, reg, evalutor);
+            exp = exp.Trim();
+            var expressions = Regex.Matches(exp, ExpressionRecognizePattern);
```

Comment:
>Regex.Matche [](start = 30, length = 12)

use new Regex("....", RegexOptions.Compiled), cache regex and use regex.Matches()